services/roomService/src/typeDefs/index.ts

```diff
@@ -8,24 +8,47 @@ type Query {
 
 type User {
   id: ID!
-  name: String
+  username: String
   email: String
 }
 
 
 input signUpInput {
-name: String !
+username: String !
 email: String
 password : String!
 }
 
 input loginInput {
   email: String!
-  pasword: String!
+  password: String!
 }
 
+input updateUsernameInput {
+  id: ID!
+  username: String
+}
+
+input updateEmailInput {
+  id: ID!
+  email: String
+}
+
+input updatePasswordInput {
+  id: ID!
+  password: String
+}
+
+input deleteUserInput {
+  email: String!
+}
 
 type Mutation {
   signUp(input: signUpInput) : User
+  login(input: loginInput) : User
+  updateUsername(input: updateUsernameInput): Boolean
+  updateEmail(input: updateEmailInput): Boolean
+  updatePassword(input: updatePasswordInput): Boolean
+  deleteUser(input: deleteUserInput): Boolean
```

Comment:
I feel , Instead of having different mutations for username / email/ password , 
we can just have one mutation called updateUser and pass username email password as optional parameters...
This would be a more optimized way.

Reply:
Got it .. fetch user was already done by me :D 
I didn't realize it .. just did that for checking setup.
So assigned update to you.